app/src/main/java/com/mk/tasky/agenda/detail/reminder/presentation/DetailReminderViewModel.kt

```diff
@@ -72,7 +134,14 @@ class DetailReminderViewModel @Inject constructor(
                 )
             }
             DetailReminderEvent.OnReminderDelete -> {
-                println("Delete Reminder")
+                state.id?.let {
+                    viewModelScope.launch(NonCancellable) {
+                        reminderUseCases.deleteReminder(it)
+                    }
+                }
+                state = state.copy(
+                    shouldExit = true
+                )
```

Comment:
Yeah, here in this case it's needed again since you exit outside of the coroutine :)

Reply:
Gotcha. Thanks for the headsup!